src/DistributedWriteAheadLog/KafkaWAL.cpp

```diff
@@ -269,7 +273,7 @@ void KafkaWAL::initProducerHandle()
         std::make_pair("message.send.max.retries", std::to_string(settings->message_send_max_retries)),
         std::make_pair("retry.backoff.ms", std::to_string(settings->retry_backoff_ms)),
         std::make_pair("enable.idempotence", std::to_string(settings->enable_idempotence)),
-        std::make_pair("compression.codec", settings->compression_codec),
+        std::make_pair("compression.codec", settings->enable_compress_internal_topic ? "none" : settings->compression_codec),
```

Comment:
`producer` is global, we need be careful here. We may turn on client compression for other production topics.